resqpy/grid/_extract_functions.py

```diff
@@ -445,7 +445,12 @@ def extract_stratigraphy(grid):
                                 object = grid,
                                 array_attribute = 'stratigraphic_units',
                                 dtype = 'int')
-    assert len(grid.stratigraphic_units) == grid.nk_plus_k_gaps
+    if (len(grid.stratigraphic_units) != grid.nk_plus_k_gaps):
+        log.warning(
+            "Number of stratigraphic units (%d) does not equal the number of layers plus gaps (%d)",
```

Comment:
We have been using the f"blah blah {expression} blah {expression}" form of string formatting elsewhere in resqpy. For consistency it might be nice to stick with that convention, though it is not particularly important, if you feel strongly about the % way of formatting!